### ISSUE TITLE
[Feature Request] Konstraint to create Constraint resource when policy uses parameters

### ISSUE THREAD

**Issue by pramodpawade** [plexsystems/konstraint#224]
## What
This is the feature request to enable the creation of Constraint resources when policy uses parameters. I know this has been there in the past but recently Konstraint removed this feature in light that this will be a user-managed field but I think if Konstraint provides this feature to just create a Constraint resources file with the necessary block of field then users can use any automation tool for example `Kustomize` to patch this file for all necessary input field.
```
Error:
WARN[0000] skipping constraint generation due to use of parameters name=Image_tag src=policy/image_tag/src.rego
```

## Why
At present, because this feature is not there, I have to generate this file manually and keep them somewhere so that I can embed this with my Constraint Template file which will be generated by Konstraint on the fly from my source rego. This becomes overhead & remove the ability to achieve end-to-end automation.

## Scenario
In my case, I am trying to utilise Konstraint to create the necessary Constraint template and Constraint on the fly from my source rego for the Gatekeeper and will use the same source rego for Conftest. At present, our workload does include a couple of Constraint template which has rego with parameters and we're expecting this list to grow. If Konstraint does manage the creation of these Constraint resources on the fly as well for parameters, I can use `Kustomize` to chip in and do the necessary patch on this file on the go as we're heavily using it for the rest of the templates by using Helm rendering.

**Comment by jalseth** [comment]
Hi @pramodpawade, thanks for opening this request. This has been discussed before in https://github.com/plexsystems/konstraint/issues/78 and at the time decided against doing this, as the primary reason to have a policy with parameters is if you intend on having users supply their own values to policies. In that scenario, the users probably don't have Konstraint (or know it exists) so I would expect them to just write the YAMLs themselves.

Please explain how you would expect Konstraint to generate the Constraint resources, including how the parameters would be defined, and how it would integrate into your workflow.

**Comment by pramodpawade** [comment]
Hi @jalseth Thank you for writing back to me.
I did checked the #78 issue but for me I think the best thing to do if parameters does exists in the source rego to create the Constraint resource with all the values except the input parameters which ideally will be pass by user. For example for source rego with parameter define as:
`# @parameter repos array string` and Konstraint will read this and create Constraint resource like below:

```
apiVersion: constraints.gatekeeper.sh/v1beta1
kind: allowedRepos
metadata:
  name: allowedRepos
spec:
  match:
    kinds:
      - apiGroups: [""]
        kinds: ["Pod"]
  parameters:
    repos:
   ```
   After this, user can easily integrate `Kustomize` tool to input the repos by using Helm render.
   
My use case here is, I am trying to use single source of truth for Conftest and Gatekeeper which will test my Helm charts before and after execution on the K8s cluster. Konstraint has all the things which I wanted here but I am also having some workloads in the Gatekeeper which utilising parameters thus make it manual to manage those Constraint resources. I want them to create on the fly by using Konstraint and then I can utilise `Kustomize` to render it and pass my input automatically.
This makes my entire solution end-to-end automated without manually managing those Constraint resources.

**Comment by jalseth** [comment]
I would be open to that if we gate it behind a flag that defaults to false. I don't think the default behavior of Konstraint should render partial, invalid Constraint resources. I'll add this and tag you in the PR for review of functionality.

**Comment by jalseth** [comment]
@pramodpawade Please take a look at #225 when you have a minute and test to see if it works for your use case.

**Comment by pramodpawade** [comment]
Hi @jalseth Thank you for writing back to me. I tested this and it is now well suited for my use case. Appreciate your quick turnaround on this!!

**Comment by jalseth** [comment]
Fixed via #225
